.github/workflows/gradle.yml

```diff
@@ -0,0 +1,69 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [8, 11, 14]
+
+    name: Java ${{ matrix.java }}
+    steps:
+    - uses: actions/checkout@v1
+    - name: Restore local Gradle & Maven cache
+      uses: actions/cache@v2.1.0
+      with:
+        path: |
+          .gradle
+          ~/.gradle
+          ~/.m2
+        key: gradle-m2-java-${{ matrix.java }}
+        restore-keys: |
+          gradle-m2-java-
+          gradle-m2
+    - name: Set up JDK ${{ matrix.java }}
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ matrix.java }}
+    - name: Assemble with Gradle
+      run: ./gradlew assemble
+    - name: Javadoc with Gradle
+      run: ./gradlew javadocs
+
+  test:
+
+    needs: build
+    runs-on: ubuntu-latest
+
+    name: Test on Java 8
+    steps:
+    - uses: actions/checkout@v1
+    - name: Restore local Gradle & Maven cache
+      uses: actions/cache@v2.1.0
+      with:
+        path: |
+          .gradle
+          ~/.gradle
+          ~/.m2
+        key: gradle-m2-java-8
+        restore-keys: |
+          gradle-m2-java-
+          gradle-m2
+    - name: Set up JDK 8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - name: Check with Gradle
+      run: ./gradlew check
```

Comment:
If we want this to be fast and not run integration tests, just add the argument:
`--exclude-task test:integration:test`

Reply:
I think we do want to run integration tests. Maybe not for all JDKs, but at least for one of them.

Reply:
`./gradlew assemble` is run against Java 11 and Java 14 in parallel, and `./gradlew check` (including integration tests) is run against only Java 11 after the assemble workflow is successful.